src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/WriteHelper.cs

```diff
@@ -919,6 +945,58 @@ public async Task<int> CopyKeyValueStoreEntryHistory(List<KeyValueStoreEntryHist
         return entities.Count;
     }
 
+    public async Task<int> CopyNonFungibleDataSchemaHistory(ICollection<NonFungibleDataSchemaHistory> nonFungibleDataSchemaEntries, CancellationToken token)
+    {
+        if (!nonFungibleDataSchemaEntries.Any())
+        {
+            return 0;
+        }
+
+        await using var writer = await _connection.BeginBinaryImportAsync("COPY non_fungible_data_schema_history (id, from_state_version, entity_id, schema, type_kind, type_index) FROM STDIN (FORMAT BINARY)", token);
+
+        foreach (var e in nonFungibleDataSchemaEntries)
+        {
+            await writer.StartRowAsync(token);
+            await writer.WriteAsync(e.Id, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.FromStateVersion, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.EntityId, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.Schema, NpgsqlDbType.Bytea, token);
+            await writer.WriteAsync(e.TypeKind, "key_type_kind", token);
+            await writer.WriteAsync(e.TypeIndex, NpgsqlDbType.Integer, token);
+        }
+
+        await writer.CompleteAsync(token);
+
+        return nonFungibleDataSchemaEntries.Count;
+    }
+
+    public async Task<int> CopyKeyValueStoreSchemaHistory(ICollection<KeyValueStoreSchemaHistory> kvSchemaHistoryEntries, CancellationToken token)
+    {
+        if (!kvSchemaHistoryEntries.Any())
+        {
+            return 0;
+        }
+
+        await using var writer = await _connection.BeginBinaryImportAsync("COPY key_value_store_schema_history (id, from_state_version, entity_id, schema, key_type_kind, key_type_index, value_type_kind, value_type_index) FROM STDIN (FORMAT BINARY)", token);
+
+        foreach (var e in kvSchemaHistoryEntries)
+        {
+            await writer.StartRowAsync(token);
+            await writer.WriteAsync(e.Id, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.FromStateVersion, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.EntityId, NpgsqlDbType.Bigint, token);
+            await writer.WriteAsync(e.Schema, NpgsqlDbType.Bytea, token);
+            await writer.WriteAsync(e.KeyTypeKind, "key_type_kind", token);
```

Comment:
Question: `key_type_kind` might be bit ambiguous, maybe we could rename it into `sbor_key_type`?

Reply:
changed type to `SborTypeKind`